tests/phpunit/tests/fonts/font-library/wpFontFamily/install.php

```diff
@@ -0,0 +1,562 @@
+						'name'     => 'lato2.ttf',
+						'type'     => 'font/ttf',
+						'tmp_name' => wp_tempnam( 'Lato-' ),
+						'error'    => 0,
+						'size'     => 123,
+					),
+				),
+				'expected'   => array( 'lato_normal_400.ttf', 'lato_normal_500.ttf' ),
+			),
+			// woff font type.
+			'woff local font'  => array(
+				'font_data'  => array(
+					'name'       => 'Cooper Hewitt',
+					'slug'       => 'cooper-hewitt',
+					'fontFamily' => 'Cooper Hewitt',
+					'fontFace'   => array(
+						array(
+							'fontFamily'   => 'Cooper Hewitt',
+							'fontStyle'    => 'italic',
+							'fontWeight'   => '900',
+							'uploadedFile' => 'files0',
+						),
+					),
+				),
+				'files_data' => array(
+					'files0' => array(
+						'name'     => 'cooper-hewitt.woff',
+						'type'     => 'font/woff',
+						'tmp_name' => wp_tempnam( 'Cooper-' ),
+						'error'    => 0,
+						'size'     => 123,
+					),
+				),
+				'expected'   => array( 'cooper-hewitt_italic_900.woff' ),
+			),
+			// woff2 font type.
+			'woff2 local font' => array(
+				'font_data'  => array(
+					'name'       => 'DM Sans',
+					'slug'       => 'dm-sans',
+					'fontFamily' => 'DM Sans',
+					'fontFace'   => array(
+						array(
+							'fontFamily'   => 'DM Sans',
+							'fontStyle'    => 'regular',
+							'fontWeight'   => '500',
+							'uploadedFile' => 'files0',
+						),
+					),
+				),
+				'files_data' => array(
+					'files0' => array(
+						'name'     => 'DMSans.woff2',
+						'type'     => 'font/woff2',
+						'tmp_name' => wp_tempnam( 'DMSans-' ),
+						'error'    => 0,
+						'size'     => 123,
+					),
+				),
+				'expected'   => array( 'dm-sans_regular_500.woff2' ),
+			),
+			// otf font type.
+			'otf local font'   => array(
+				'font_data'  => array(
+					'name'       => 'Gilbert Color',
+					'slug'       => 'gilbert-color',
+					'fontFamily' => 'Gilbert Color',
+					'fontFace'   => array(
+						array(
+							'fontFamily'   => 'Gilbert Color',
+							'fontStyle'    => 'regular',
+							'fontWeight'   => '500',
+							'uploadedFile' => 'files0',
+						),
+					),
+				),
+				'files_data' => array(
+					'files0' => array(
+						'name'     => 'gilbert-color.otf',
+						'type'     => 'application/vnd.ms-opentype',
+						'tmp_name' => wp_tempnam( 'Gilbert-' ),
+						'error'    => 0,
+						'size'     => 123,
+					),
+				),
+				'expected'   => array( 'gilbert-color_regular_500.otf' ),
+			),
+		);
+	}
+
+	/**
+	 * @dataProvider data_should_not_install_duplicate_fontfaces
+	 *
+	 * @param array $font_data  Font family data in theme.json format.
+	 * @param array $files_data Files data in $_FILES format.
+	 * @param array $expected   Expected font filename(s).
+	 */
+	public function test_should_not_install_duplicate_fontfaces( $font_data, array $files_data, array $expected ) {
+		// Set up the temporary files.
+		foreach ( $files_data as $file ) {
+			copy( __DIR__ . '/../../../data/fonts/Merriweather.ttf', $file['tmp_name'] );
```

Comment:
Test directory const

Reply:
Thanks for flagging that. Done here: https://github.com/WordPress/wordpress-develop/commit/e218ec8b94e679da12c013bf7ce617f169ee0738